commit 867a4f06386e19d6057afb4918bafba809d900a9
Author: Zach Ploskey <zach@ploskey.com>
Date:   Sun Jan 22 00:47:19 2017 -0800

    platform/x86: ideapad-laptop: handle ACPI event 1
    
    
    [ Upstream commit cfee5d63767b2e7997c1f36420d008abbe61565c ]
    
    On Ideapad laptops, ACPI event 1 is currently not handled. Many models
    log "ideapad_laptop: Unknown event: 1" every 20 seconds or so while
    running on battery power. Some convertible laptops receive this event
    when switching in and out of tablet mode.
    
    This adds and additional case for event 1 in ideapad_acpi_notify to call
    ideapad_input_report(priv, vpc_bit), so that the event is reported to
    userspace and we avoid unnecessary logging.
    
    Fixes bug #107481 (https://bugzilla.kernel.org/show_bug.cgi?id=107481)
    Fixes bug #65751 (https://bugzilla.kernel.org/show_bug.cgi?id=65751)
    
    Signed-off-by: Zach Ploskey <zach@ploskey.com>
    Signed-off-by: Andy Shevchenko <andriy.shevchenko@linux.intel.com>
    Signed-off-by: Sasha Levin <alexander.levin@verizon.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

diff --git a/drivers/platform/x86/ideapad-laptop.c b/drivers/platform/x86/ideapad-laptop.c
index a7614fc..2f1615e 100644
--- a/drivers/platform/x86/ideapad-laptop.c
+++ b/drivers/platform/x86/ideapad-laptop.c
@@ -813,6 +813,7 @@ static void ideapad_acpi_notify(acpi_handle handle, u32 event, void *data)
 			case 8:
 			case 7:
 			case 6:
+			case 1:
 				ideapad_input_report(priv, vpc_bit);
 				break;
 			case 5:
